Build page on /solutions/cloud-native-development

minkyngkm opened the pull request:
## Done
- Build page on `/solutions/cloud-native-development`
- Build a get in touch modal 
## QA

- Go to  https://canonical-com-1577.demos.haus/solutions/cloud-native-development
- Compare page against [Figma](https://www.figma.com/design/brgzRViEncgKKD4KzVnyXy/canonical.com%2Fsolutions?node-id=2212-4394&t=GzX8JZVNlu5WJdgc-0) and [copy doc](https://docs.google.com/document/d/1xn9QcJ7OfELzRBtUlu-WkSSnC9lYideoUu-YV1yBNIU/edit?tab=t.30h92ifbdc1)
- Check the get in touch modal works fine 


## Issue / Card

Fixes #https://warthogs.atlassian.net/browse/WD-19255

minkyngkm commented:
Hi @kim-isaac and @eliman11, this PR is ready for UX and design review. Please feel free to take a look at the [demo](https://canonical-com-1577.demos.haus/solutions/cloud-native-development) when you are available. Thanks!

kim-isaac commented:
Thanks @minkyngkm, I have some comments below:
You can check full comments on [Figma](https://www.figma.com/design/brgzRViEncgKKD4KzVnyXy?node-id=2120-4995#1150203850).
![Screenshot 2025-02-26 at 17 52 46](https://github.com/user-attachments/assets/907141d3-9eab-4c28-89f3-1b55673ebe58)

- [ ] As I mentioned in [Jira](https://warthogs.atlassian.net/browse/WD-19255?focusedCommentId=688133), this part should be hidden for now.

![Screenshot 2025-02-26 at 17 54 25](https://github.com/user-attachments/assets/e7791ad7-079e-4e55-b943-c3d843cb9aeb)

- [ ] For this part, could we use 2-4 column instead of 3-3?

**-For tablet size**

![Screenshot 2025-02-26 at 17 55 54](https://github.com/user-attachments/assets/2b913181-15a5-445d-844f-2240a910203f)

- [ ] For all blue highlighted sections(let us... / deply.../run your.../one subscription.../choose your...); could we use 50:50 with image like this [telco/open-ran page](https://canonical.com/solutions/telco/open-ran)?

**-For mobile**

![Screenshot 2025-02-26 at 17 59 47](https://github.com/user-attachments/assets/736419da-5574-4b57-830a-92734a8bbcbb)

- [ ] (seems like)Image container didn’t fully removed here, making additional margin. Could you remove it?
(from this to next section)
- [ ] Also, can we skip 24px bottom margin like ‘Choose your app modernization journey’ section?
(from this to ‘One subscription. No surprises.’ section)

![Screenshot 2025-02-26 at 18 00 44](https://github.com/user-attachments/assets/39b79e5e-157b-48a8-9754-16a4d619f645)

- [ ] For all quote sections, could you remove bottom margin? (example below)

Thanks again and let me know if you have any questions!

minkyngkm commented:
Thanks @kim-isaac and @eliman11 for reviewing. I've addressed all your comments. Hopefully it looks better now! https://canonical-com-1577.demos.haus/solutions/cloud-native-development

kim-isaac commented:
Thanks for addressing things @minkyngkm , It looks great! A few comments left for tablet size:
![Screenshot 2025-02-27 at 15 14 39](https://github.com/user-attachments/assets/982b0a80-eb8e-477c-a8f4-ddfc176a8aa1)

- [ ] Could we have the images also for Tablet?

![Screenshot 2025-02-27 at 15 16 17](https://github.com/user-attachments/assets/2a6a0648-cb35-4c83-ad6d-625801f2c462)

- [ ] Could you make this a 50:50 layout as well?

That's all from visual side, thanks again!

minkyngkm commented:
@kim-isaac sorry, all done! 🙏 https://canonical-com-1577.demos.haus/solutions/cloud-native-development

petesfrench commented on static/js/modals.js:
```diff
@@ -309,3 +309,26 @@ function getCustomFields(event) {
+    tab.addEventListener("click", function (event) {
+      event.preventDefault();
+
+      document.querySelectorAll(".p-tabs__link").forEach((t) => {
```
Is it needed to make the `querySelectorAll` a second time here?

petesfrench commented on static/js/modals.js:
```diff
@@ -309,3 +309,26 @@ function getCustomFields(event) {
 }
+
+// Fix Tab and modal js conflicts
+(function () {
```
Haven't looked into it, but would it be possible to add a check to the initial ['aria-controls' listener for forms](https://github.com/canonical/canonical.com/blob/main/static/js/modals.js#L138) to see if it has an associated form (with `document.getElementById(targetControls)`) before calling `toggleModal`, so it doesn't affect other elements with aria-controls?
